addons/purchase/views/portal_templates.xml

```diff
@@ -257,10 +296,11 @@
                             <th class="text-start">Products</th>
                             <th class="text-end">Quantity</th>
                             <th t-if="update_dates" class="text-end">Scheduled Date</th>
-                            <th t-if="not update_dates and order.state in ['purchase', 'done']" t-attf-class="text-end {{ 'd-none d-sm-table-cell' if report_type == 'html' else '' }}">Unit Price</th>
-                            <th t-if="not update_dates and order.state in ['purchase', 'done']" t-attf-class="text-end {{ 'd-none d-md-table-cell' if report_type == 'html' else '' }}">
+                            <th t-if="not update_dates and order.state in ['sent', 'purchase', 'done']" t-attf-class="text-end {{ 'd-none d-sm-table-cell' if report_type == 'html' else '' }}">Unit Price</th>
+                            <th t-if="not update_dates and order.state in ['sent', 'purchase', 'done']" t-attf-class="text-end {{ 'd-none d-md-table-cell' if report_type == 'html' else '' }}">
                                 <span>Taxes</span>
                             </th>
+                            <th t-if="not update_dates and order.state in ['sent', 'purchase', 'done']" t-attf-class="text-end {{ 'd-none d-sm-table-cell' if report_type == 'html' else '' }}">Disc.%</th>
```

Comment:
Not related to your change but the `<th>` for "Amount" and its `<td>` don't have the same condition:
- `order.state in ['purchase', 'done']` for the `<th>`
- `not update_dates and order.state in ['purchase', 'done']` for the `<td>`
... which means `<th>` will be displayed if `update_dates` is true but the `<td>` will not. Can you check that ?